docs/DeveloperGuide.adoc

```diff
@@ -153,94 +153,234 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+
+[start=2]
+. Searching by *area*
+* An *area* refers to one of the five areas of Singapore
+** Central
+** East
+** North East
+** West
+** North
+* You can obtain more detailed information about each area from
+https://keylocation.sg/singapore/districts-map[this website]
+
+The nearby feature was implemented as a `NearbyCommand` in the `Logic` package. +
+
+==== Class Diagram for Nearby feature
+.Nearby Command Class Diagram
+image::NearbyClassDiagram.svg[]
+
+==== Sequence Diagram for searching by postal sector
+The sequence diagram for the *Nearby Command* when searching by *postal sector*
+is shown below:
+
+.Nearby Command Sequence Diagram
+image::NearbyCommandSequenceDiagram1.svg[]
+
+The arguments passed to the Nearby Command will be parsed by the `NearbyCommandParser` class. +
+If the given arguments are valid, a new `NearbyCommand` object will be returned.
+In this class, invalid arguments will result in a `ParseException`. +
+Two types of invalid arguments are checked for: empty arguments
+and arguments with only whitespace characters.
+
+The `execute()` function of the `NearbyCommand` will first check if the given arguments
+are in the format required for postal sector search (the argument can be converted into an integer).
+The check is done by the `isPostalSectorSearch()` function. +
+If the first check is successful, the argument will be converted into an integer and
+a second check is performed via the `isValidPostalSector(Index postalSector)` function
+of the `NearbyCommandUtil` helper class.
 
 [NOTE]
-If a command fails its execution, it will not call `Model#commitAddressBook()`, so the address book state will not be saved into the `addressBookStateList`.
+====
+`NearbyCommandUtil` is a helper class that contains functions and variables used for
+identifying postal sectors and their corresponding general locations. +
+This class was created to reduce the responsibility of the `NearbyCommand` class. +
+
+* A `HashMap` was used to store information about postal sectors and their respective general locations.
+====
+
+The `showPostalSectors(Model model, Index postalSector)` function of the `NearbyCommand` is called.
+This function updates the model's filtered order list using the predicate obtained from the
+`getPostalSectorPredicate()` function.
+
+A `CommandResult` is then generated and returned to the `LogicManager`.
 
-Step 4. The user now decides that adding the person was a mistake, and decides to undo that action by executing the `undo` command. The `undo` command will call `Model#undoAddressBook()`, which will shift the `currentStatePointer` once to the left, pointing it to the previous address book state, and restores the address book to that state.
+==== Sequence Diagram for searching by area
+The sequence diagram for the *Nearby Command* when searching by *area* is shown below:
 
-image::UndoRedoState3.png[]
+.Nearby Command Sequence Diagram 2
+image::NearbyCommandSequenceDiagram2.svg[]
```

Comment:
The two Nearby Command Sequence Diagram contains similar elements and interaction working process, with minor changes, it might be considered to keep only one of them. 😄